tests/undocommand.js

```diff
@@ -0,0 +1,310 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Editor from '/ckeditor5/editor.js';
+import ModelDocument from '/ckeditor5/engine/treemodel/document.js';
+import Range from '/ckeditor5/engine/treemodel/range.js';
+import Position from '/ckeditor5/engine/treemodel/position.js';
+import UndoCommand from '/ckeditor5/undo/undocommand.js';
+
+let element, editor, doc, root, undo;
+
+beforeEach( () => {
+	element = document.createElement( 'div' );
+	document.body.appendChild( element );
+
+	editor = new Editor( element );
+	undo = new UndoCommand( editor );
+
+	doc = new ModelDocument();
+	editor.document = doc;
+
+	root = doc.createRoot( 'root' );
+} );
+
+afterEach( () => {
+	undo.destroy();
+} );
+
+describe( 'UndoCommand', () => {
+	describe( 'constructor', () => {
+		it( 'should create undo command with empty batch stack', () => {
+			expect( undo._batchStack.length ).to.equal( 0 );
+		} );
+	} );
+
+	describe( 'addBatch', () => {
+		it( 'should add a batch and selection to command stack', () => {
+			root.appendChildren( 'foobar' );
+
+			editor.document.selection.setRanges( Range.createFromElement( root ) );
+
+			const batch = doc.batch();
+			undo.addBatch( batch );
+
+			expect( undo._batchStack.length ).to.equal( 1 );
+			expect( undo._batchStack[ 0 ] ).to.equal( batch );
+
+			expect( undo._batchSelection.get( batch ) ).to.deep.equal( Array.from( editor.document.selection.getRanges() ) );
+		} );
+	} );
+
+	describe( 'clearStack', () => {
+		it( 'should remove all batches from the stack', () => {
+			undo.addBatch( doc.batch() );
+			undo.clearStack();
+
+			expect( undo._batchStack.length ).to.equal( 0 );
+			expect( undo._batchSelection.size ).to.equal( 0 );
+		} );
+	} );
+
+	describe( '_checkEnabled', () => {
+		it( 'should return false if there are no batches in command stack', () => {
+			expect( undo._checkEnabled() ).to.be.false;
+		} );
+
+		it( 'should return true if there are batches in command stack', () => {
+			undo.addBatch( doc.batch() );
+
+			expect( undo._checkEnabled() ).to.be.true;
+		} );
+	} );
+
+	describe( '_doExecute', () => {
```

Comment:
There should not be tests for private class. If you want to test this method you should change it to "protected".